src/processTargets/modifiers/SmallPaintStage.ts

```diff
@@ -0,0 +1,66 @@
+import { zip } from "lodash";
+import { Target } from "../../typings/target.types";
+import {
+  ContainingScopeModifier,
+  EveryScopeModifier,
+} from "../../typings/targetDescriptor.types";
+import { ProcessedTargetsContext } from "../../typings/Types";
+import { ModifierStage } from "../PipelineStages.types";
+import { TokenTarget } from "../targets";
+import { NonWhitespaceSequenceStage } from "./scopeTypeStages/RegexStage";
+import { InteriorOnlyStage } from "./InteriorStage";
+
+export type NonWhitespaceOrQuoteSequenceModifier = (
+  | ContainingScopeModifier
+  | EveryScopeModifier
+) & {
+  scopeType: { type: "nonWhitespaceOrQuoteSequence" };
+};
+
+/**
+ * Intersection of NonWhitespaceSequenceStage and InteriorOnlyStage
+ * Expand the target until reaching a white space or surrounding pair.
+ * If there is no surrounding pair defaults to the non white space sequence
+ */
+export default class SmallPaintStage implements ModifierStage {
+  constructor(private modifier: NonWhitespaceOrQuoteSequenceModifier) {}
+
+  run(context: ProcessedTargetsContext, target: Target): Target[] {
+    const paintStage = new NonWhitespaceSequenceStage({
+      type: "containingScope",
+      scopeType: { type: "nonWhitespaceSequence" },
+    });
+
+    const surroundingPairStage = new InteriorOnlyStage({
+      type: "interiorOnly",
+    });
+
+    const paintTargets = paintStage.run(context, target);
+
+    try {
+      const pairTargets = surroundingPairStage.run(context, target);
```

Comment:
Instead of using the stage we could probably just call the function directly
https://github.com/cursorless-dev/cursorless/blob/256d656b2e84ebc6d29ab9ee5515f3eee396fb2d/src/processTargets/modifiers/SurroundingPairStage.ts#L37

Just check if `pairInfo` is not null and then do an intersection with `pairInfo.interiorRange`

Reply:
As above, I'd be inclined not to tightly couple where possible, so I'd lean towards using a stage.  But maybe this simplifies things somehow?

Reply:
I've gone with this suggestion because it does make the code simpler to write and understand.